fastly/block_fastly_service_v1_logging_elasticsearch.go

```diff
@@ -0,0 +1,284 @@
+package fastly
+
+import (
+	"fmt"
+	"log"
+
+	fst "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var elasticsearchSchema = &schema.Schema{
+	Type:     schema.TypeSet,
+	Optional: true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			// Required fields
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique name of the Elasticsearch logging endpoint",
+			},
+
+			"url": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The Elasticsearch URL to stream logs to.",
+			},
+
+			"index": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the Elasticsearch index to send documents (logs) to.",
+			},
+
+			// Optional fields
+			"pipeline": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The ID of the Elasticsearch ingest pipeline to apply pre-process transformations to before indexing.",
+			},
+
+			"user": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "BasicAuth user.",
+			},
+
+			"password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "BasicAuth password.",
+				Sensitive:   true,
+			},
+
+			"request_max_entries": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The maximum number of logs sent in one request.",
+			},
+
+			"request_max_bytes": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The maximum number of bytes sent in one request.",
```

Comment:
@ezkl and I talked offline. This is validated by the API and the failure mode is handled nicely. Therefore, to avoid the possibility of validation drift, omitting the validation here.